codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithydafny/DafnyApiCodegen.java

```diff
@@ -349,12 +349,28 @@ public TokenTree generateEnumTypeDefinition(final ShapeId stringShapeId) {
     if (!enumTrait.hasNames()) {
       throw new UnsupportedOperationException("Unnamed enums not supported");
     }
+
+    // Dafny uses the enum's name, rather than its value here.
+    // Python (and possibly other runtimes) use values.
```

Comment:
Do we know for sure that we should switch Dafny codegen from using names to using values, and not switch the runtime codegen from using values to using names?

Reply:
`ComAmazonawsS3Types.py` uses the value, not the name. As far as I know we can't change that within Smithy-Dafny, since it's generated by Dafny's `translate`. I'll also ask Robin about it, since his comment on https://github.com/smithy-lang/smithy-dafny/issues/780 implies that `smithy-dafny` prefers to use the name.